frontend/src/features/logic/utils/getVisibleFieldIds.ts

```diff
@@ -0,0 +1,205 @@
+import { intersection } from 'lodash'
+import { CamelCasedProperties } from 'type-fest'
+
+import { BasicField, FormFieldDto } from '~shared/types/field'
+import { FormCondition, FormDto, LogicConditionState } from '~shared/types/form'
+
+import { CHECKBOX_OTHERS_INPUT_VALUE } from '~templates/Field/Checkbox/CheckboxField'
+import { RADIO_OTHERS_INPUT_VALUE } from '~templates/Field/Radio/RadioField'
+
+import {
+  validateCheckboxInput,
+  validateDateInput,
+  validateRadioInput,
+  validateSingleAnswerInput,
+  validateVerifiableInput,
+} from '~features/public-form/utils'
+
+import { FieldIdSet, FieldIdToType } from '../types'
+
+import { groupLogicUnitsByField } from './groupLogicUnitsByField'
+
+const getCurrentFieldValue = (input: unknown, fieldType: BasicField) => {
+  switch (fieldType) {
+    case BasicField.Checkbox:
+      return validateCheckboxInput(input) ? input : null
```

Comment:
do we really need to validate in this function? doesn't this mean we are running validation every single time any change is made to the form?

Reply:
not just validation but we are re-validating outputs which are completely in our own control

Reply:
As mentioned in https://github.com/opengovsg/FormSG/pull/3597#discussion_r843505083, will refactor this to use a cast instead.